pkg/secrets/secrets.go

```diff
@@ -0,0 +1,92 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/redpanda-data/common-go/secrets"
+)
+
+type CloudSecretsExpander struct {
+	client secrets.SecretAPI
+}
+
+// NewCloudSecretsExpander creates a new CloudSecretsExpander
+func NewCloudSecretsExpander(
+	ctx context.Context,
+	prefix string,
+	awsRegion string,
+	awsRoleARN string,
+	gcpProjectID string,
+	azureKeyVaultURI string,
+	logger *slog.Logger,
+) (*CloudSecretsExpander, error) {
+	var secretsAPI secrets.SecretAPI
+	var err error
+	if awsRegion != "" {
+		secretsAPI, err = secrets.NewAWSSecretsManager(
+			ctx,
+			logger,
+			awsRegion,
+			awsRoleARN,
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else if gcpProjectID != "" {
+		secretsAPI, err = secrets.NewGCPSecretsManager(
+			ctx,
+			logger,
+			gcpProjectID,
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else if azureKeyVaultURI != "" {
+		secretsAPI, err = secrets.NewAzSecretsManager(
+			logger,
+			azureKeyVaultURI,
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("incorrect cloud secret store configuration")
+	}
+	provider, _ := secrets.NewSecretProvider(secretsAPI, prefix, "")
```

Comment:
Why are we discarding this error and what's the `""`? 🤔 Could you denote the answers of that in a comment please?

Reply:
I'd be okay with a comment saying "This never returns an error: <Link to code>".

Reply:
I handled the error. I think it's better to not make assumptions since the implementation can change